### PR TITLE
Add Cohere Chat blueprint with RAG

### PR DESCRIPTION
**Pull request by tianjing-li**
### Description
[Describe what this change achieves]

- Adds a new Cohere Chat connector blueprint
 
### Issues Resolved
[List any issues this PR will resolve]
N/A
 
### Check List
No new functionality, just new docs 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tianjing-li** [comment]
@ylwu-amzn also updated this new doc with some of the suggestions from the Embed one. Take a look when you get a chance, many thanks.

File: docs/remote_inference_blueprints/cohere_connector_chat_blueprint.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+### Cohere Chat Connector Blueprint:
+
+This blueprint will show you how to connect a Cohere chat model to your Opensearch cluster. You will require a Cohere API key to create a connector.
+
+It is suggested to use the default `command` model for using Chat. Cohere's Chat endpoint also features Retrievel Augmented Generation (or RAG) parameters, by adding `connectors` or `documents` in your request body, in addition to allowing you to pass a `conversation_id` to provide context to Cohere's model.
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/cohere_connector_chat_blueprint.md:5]: > by adding `connectors` or `documents` in your request body

I think you mean Cohere connectors here, right?

Comment (tianjing-li) [reply docs/remote_inference_blueprints/cohere_connector_chat_blueprint.md:5]: Yes correct, Cohere connectors - there should be text earlier linking docs and explaining the difference, but I can clarify that here as well

File: docs/remote_inference_blueprints/cohere_connector_chat_blueprint.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+You can try this request to test that the Model behaves correctly:
+
+Note: `conversation_id` here is the string value for "1", but you can use any string value, such as generating a uuidv4 string. This value will be org-specific and you will need to manage it on your application side.
+
+```json
+POST /_plugins/_ml/models/<MODEL_ID_HERE>/_predict
+{
+  "parameters": {
+    "message": "What is the weather like in London?",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/cohere_connector_chat_blueprint.md:129]: Make sure the example in this doc can work
```suggestion
    "message": "What is the weather like in London?"
```

File: docs/remote_inference_blueprints/cohere_connector_chat_blueprint.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://api.cohere.ai/v1/chat",
+      "headers": {
+        "Authorization": "Bearer ${credential.cohere_key}",
+        "Request-Source": "unspecified:opensearch"
+      },
+      "request_body": "{ \"message\": ${parameters.message}, \"model\": \"${parameters.model}\" }"
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/cohere_connector_chat_blueprint.md:59]:
```suggestion
      "request_body": "{ \"message\": \"${parameters.message}\", \"model\": \"${parameters.model}\" }"
```

**Review by ylwu-amzn — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

**Comment by ylwu-amzn** [comment]
@tianjing-li  You can refer to this tutorial for how to use Cohere command model in RAG pipeline https://github.com/opensearch-project/ml-commons/pull/2075

**Comment by tianjing-li** [comment]
@ylwu-amzn thank you!

**Pull request merged**